go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -177,13 +178,55 @@ func (p DoltDatabaseProvider) Database(ctx *sql.Context, name string) (db sql.Da
 		return nil, err
 	}
 	if !ok {
-		return nil, sql.ErrDatabaseNotFound.New(name)
+		db, err = p.databaseForClone(ctx, name)
+		if err != nil {
+			// TODO: not found here?
+			return nil, err
+		}
+
+		if db == nil {
+			return nil, sql.ErrDatabaseNotFound.New(name)
+		}
 	}
 
 	// Don't track revision databases, just instantiate them on demand
 	return db, nil
 }
 
+// attemptCloneReplica attempts to clone a database from the configured replication remote URL template, returning an error
+// if it cannot be found
+// TODO: distinct error for not found v. others
+func (p DoltDatabaseProvider) attemptCloneReplica(ctx *sql.Context, dbName string) error {
+	// TODO: these need some reworking, they don't make total sense together
+	_, readReplicaRemoteName, _ := sql.SystemVariables.GetGlobal(dsess.ReadReplicaRemote)
+	if readReplicaRemoteName == "" {
+		// not a read replica DB
+		return nil
+	}
+
+	remoteName := readReplicaRemoteName.(string)
+
+	// TODO: error handling when not set
+	_, remoteUrlTemplate, _ := sql.SystemVariables.GetGlobal(dsess.ReplicationRemoteURLTemplate)
+	if remoteUrlTemplate == "" {
+		return nil
+	}
+
+	// TODO: url sanitize
+	// TODO: SQL identifiers aren't case sensitive, but URLs are, need a plan for this
+	remoteUrl := fmt.Sprintf(remoteUrlTemplate.(string), dbName)
```

Comment:
Right now the URL template has a printf marker (%s) in it, but that's probably not what we want.

Maybe `$database$`? Thoughts?

Reply:
I went with {database}, thanks!